src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1293,1364 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+    } else {
+        //
+    }
+    // Load (water) side temperature variable
+    Real64 waterTempforCurve = this->loadSideInletTemp;
+    if (this->waterTempCurveInputVar == WaterTempCurveVar::LeavingCondenser || this->waterTempCurveInputVar == WaterTempCurveVar::LeavingEvaporator) {
+        waterTempforCurve = this->loadSideOutletTemp;
+    } else {
+        //
+    }
+
+    // 2022-05-17: should the following curve evaluation based on the oaVariable and waterVariable choice:?
+    // 2022-05-17: Maybe how this is set up is related to the flow mode?
+    // evaluate capacity modifier curve and determine load side heat transfer
+    Real64 capacityModifierFuncTemp =
+        // CurveManager::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+        // CurveManager::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, oaTempforCurve);
+        Curve::CurveValue(state, this->capFuncTempCurveIndex, waterTempforCurve, oaTempforCurve);
+
+    if (capacityModifierFuncTemp < 0.0) {
+        if (this->capModFTErrorIndex == 0) {
+            ShowSevereMessage(state, format("{} \"{}\":", DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)], this->name));
+            ShowContinueError(state,
+                              format(" Capacity Modifier curve (function of Temperatures) output is negative ({:.3T}).", capacityModifierFuncTemp));
+            ShowContinueError(state,
+                              format(" Negative value occurs using a water temperature of {:.2T}C and an outdoor air temperature of {:.2T}C.",
+                                     waterTempforCurve,
+                                     oaTempforCurve));
+            ShowContinueErrorTimeStamp(state, " Resetting curve output to zero and continuing simulation.");
+        }
+        ShowRecurringWarningErrorAtEnd(state,
+                                       format("{} \"{}\": Capacity Modifier curve (function of Temperatures) output is negative warning continues...",
+                                              DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)],
+                                              this->name),
+                                       this->capModFTErrorIndex,
+                                       capacityModifierFuncTemp,
+                                       capacityModifierFuncTemp);
+        capacityModifierFuncTemp = 0.0;
+    }
+
+    Real64 availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
+    Real64 partLoadRatio = 0.0;
+    if (availableCapacity > 0) {
+        partLoadRatio = max(0.0, min(std::abs(FFHPloadSideLoad) / availableCapacity, 1.0));
+    }
+
+    // evaluate the actual current operating load side heat transfer rate
+
+    // this->loadSideHeatTransfer = availableCapacity * partLoadRatio;
+    this->loadSideHeatTransfer = availableCapacity * (partLoadRatio >= this->minPLR ? partLoadRatio : 0.0);
```

Comment:
This seems wrong. `partLoadRatio < minPLR` says it is cycling and there are corrections for cycling later in this function, so the load shouldn't be set to zero when cycling.

Reply:
@jcyuan2020 any comment here?

Reply:
A recent commit addressed this. The current resolution is to use the plain `partLoadRatio` here. There are a few places after this will limits the `partLoadRatio` to be in proper ranges according to inputs and calculations.